velox/exec/QueryDataWriter.cpp

```diff
@@ -42,6 +42,11 @@ void QueryDataWriter::write(const RowVectorPtr& rows) {
   if (FOLLY_UNLIKELY(finished_)) {
     return;
   }
+  if (dataType_ == nullptr) {
+    dataType_ = rows->type();
+  } else {
+    VELOX_CHECK(dataType_->kindEquals(rows->type()));
```

Comment:
Should we remove this check and use `FOLLY_UNLIKELY(dataType_ == nullptr)` as this is a performance critical path inside `Operator::addInput`?

Reply:
It's actually not that bad as we call it once per batch. But yeah sure we can remove it and have what you mentioned above